modules/extension/wmts/src/main/java/org/geotools/ows/wmts/map/WMTSCoverageReader.java

```diff
@@ -664,48 +677,31 @@ private GridCoverage2D getMapReproject(
     CoordinateReferenceSystem getBestSourceCRS(
             CoordinateReferenceSystem targetCRS, WMTSReadParameters readParameters)
             throws IOException {
-        try {
-            CoordinateReferenceSystem result = null;
-            String switched = switchDefinition(targetCRS);
-            if (switched != null) {
-                // check first by changing the srs definition
-                result = CRS.decode(switched);
-                LOGGER.fine(() -> "Will request tiles with server supported crs " + switched);
-            } else if (readParameters.getSourceCRS() != null) {
```

Comment:
Removing support for user-provided CRS? 
It has been added on purpose to deal with poorly configured WMTS services, and to allow the WMTS client to decide which default CRS they prefer to use. It has nothing to do with removing a "random supported CRS", this is not random, it's a user decision.